src/Promise.js

```diff
@@ -196,6 +196,20 @@ Promise.prototype.always = function (fn) {
   return this.then(fn, fn);
 };
 
+/**
+ * Execute given callback when the promise either resolves or rejects.
+ * Same semantics as Node's Promise.finally()
+ * @param {Function} fn
+ * @returns {Promise} promise
+ */
+Promise.prototype.finally = function (fn) {
+  const onFinally = callback => Promise.resolve(fn()).then(callback);
```

Comment:
@wmertens This will not work with the `Promise` implementation used here as `Promise.resolve` is not implemented. I think we can do something like

```js
const onFinally = (callback) => {
  const wrapper = Promise.defer();
  wrapper.promise.then(callback);
  wrapper.resolve(fn());
  return wrapper.promise;
}
```

Reply:
Ah indeed - I gave you access to my fork, feel free to add commits to this branch, unfortunately I'm low on time for the moment 🙏

Reply:
I ended up with an implementation
```javascript
Promise.prototype.finally = function (fn) {
  var me = this;
  const res = function() {
    return me;
  }
  const final = () => {
    const wrapper = Promise.defer();
    wrapper.promise.then(res);
    wrapper.resolve(fn());
    return wrapper.promise;
  };

  return this.then(final, final);
}
```

This allows `finally` to execute if the promises is `resolved` or `rejected` and allows for the promise chain to continue. If you are okay with this I can push to the branch along with some changes to unit tests.